From b1be5af3d240cc8f23c8a7d774e3042ac9668b33 Mon Sep 17 00:00:00 2001
From: Felix <none@none.none>
Date: Wed, 7 Nov 2018 09:37:35 +0100
Subject: [PATCH] Tone: use DRM for display

---
 update_defconfig.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/update_defconfig.sh b/update_defconfig.sh
index ee89f33..cb09359 100755
--- a/update_defconfig.sh
+++ b/update_defconfig.sh
@@ -37,7 +37,7 @@ loire)
     DRM="false";;
 tone)
     DEVICE=$TONE;
-    DRM="false";;
+    DRM="true";;
 yoshino)
     DEVICE=$YOSHINO;
     DRM="false";;
-- 
2.19.1

