From c7abc064c21efbf79e217e2578086a4707364f97 Mon Sep 17 00:00:00 2001
From: Felix <none@none.none>
Date: Wed, 9 Jan 2019 14:04:00 +0100
Subject: [PATCH] Enable development settings by default

---
 .../settingslib/development/DevelopmentSettingsEnabler.java  | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/packages/SettingsLib/src/com/android/settingslib/development/DevelopmentSettingsEnabler.java b/packages/SettingsLib/src/com/android/settingslib/development/DevelopmentSettingsEnabler.java
index d21da4e1cd0..168a3d3c8bf 100644
--- a/packages/SettingsLib/src/com/android/settingslib/development/DevelopmentSettingsEnabler.java
+++ b/packages/SettingsLib/src/com/android/settingslib/development/DevelopmentSettingsEnabler.java
@@ -40,9 +40,10 @@ public class DevelopmentSettingsEnabler {
 
     public static boolean isDevelopmentSettingsEnabled(Context context) {
         final UserManager um = (UserManager) context.getSystemService(Context.USER_SERVICE);
+        // frameworks/base/core/java/android/provider/Settings.java
+        // getInt(ContentResolver, name, default value)
         final boolean settingEnabled = Settings.Global.getInt(context.getContentResolver(),
-                Settings.Global.DEVELOPMENT_SETTINGS_ENABLED,
-                Build.TYPE.equals("eng") ? 1 : 0) != 0;
+                Settings.Global.DEVELOPMENT_SETTINGS_ENABLED, 1) != 0;
         final boolean hasRestriction = um.hasUserRestriction(
                 UserManager.DISALLOW_DEBUGGING_FEATURES);
         final boolean isAdminOrDemo = um.isAdminUser() || um.isDemoUser();
-- 
2.20.1

