diff --git a/arch/arm64/boot/dts/qcom/msm8996-tone-common.dtsi b/arch/arm64/boot/dts/qcom/msm8996-tone-common.dtsi
index 2eff1df2bae5..17790f4d024b 100644
--- a/arch/arm64/boot/dts/qcom/msm8996-tone-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8996-tone-common.dtsi
@@ -279,8 +279,8 @@
 			preset_x_max = <1079>;
 			preset_y_max = <1919>;
 			preset_n_fingers = <10>;
-			wakeup_gesture_supported = <0>;
-			wakeup_gesture_lpm_disabled = <0>;
+			wakeup_gesture_supported = <1>;
+			wakeup_gesture_lpm_disabled = <1>;
 			wakeup_gesture_timeout = <0>;
 			wakeup_gesture {
 				double_tap {
